apps/user/models/user.go

```diff
@@ -26,6 +39,20 @@ type User struct {
 	Profiles           []UserProfile
 }
 
+//BeforeSave ...
+func (user *User) BeforeSave() {
+	var timeconfig []TimeProviderConfig
+
+	user.FirstName = strings.TrimSpace(user.FirstName)
+	user.LastName = strings.TrimSpace(user.LastName)
+	user.Email = strings.TrimSpace(user.Email)
+
+	json.Unmarshal([]byte(user.TimeProviderConfig), &timeconfig)
+	timeconfig[0].Data.Email = strings.TrimSpace(timeconfig[0].Data.Email)
```

Comment:
do this for all time providers instead of first.

Reply:
done